### PR TITLE
feat: add scheduler, and scheduler_options

### PR DESCRIPTION
**Pull request by guenthermi**
feat: add scheduler, and scheduler_options
---

- [x] This PR references an open issue
- [x] I have added a line about this change to CHANGELOG

### CODE REVIEW

File: docs/walkthrough/run-job.md [hunk 1]
```diff
@@ -134,8 +136,19 @@ Otherwise, it could happen, that your model overfits on the training data and fo
 Similarly, two or three epochs (number of passes thorough the training data) are often enough for a fine-tuning job. 
 ```
 
-
-### Construction of Training Batches
+### Configuration of a learning rate scheduler
```
Comment (gmastrapas) [inline docs/walkthrough/run-job.md:139]:
```suggestion
### Configuration of a learning rate scheduler

```

File: finetuner/__init__.py [hunk 2]
```diff
@@ -191,8 +192,12 @@ def fit(
     :param batch_size: Number of items to include in a batch.
     :param callbacks: List of callback stub objects.
         subpackage for available options, or run `finetuner.list_callbacks()`.
-    :param scheduler_step: At which interval should the learning rate scheduler's
-        step function be called. Valid options are `batch` and `epoch`.
+    :param scheduler: Name of a scheduler to use for learning rate scheduling.
+        Supported types are: `linear`, `cosine`, `cosine_with_restarts`, `polynomial`,
+        `constant`, `constant_with_warmup`
```
Comment (gmastrapas) [inline finetuner/__init__.py:197]:
```suggestion
        `constant`, `constant_with_warmup`.
```

File: finetuner/__init__.py [hunk 3]
```diff
@@ -191,8 +192,12 @@ def fit(
...
+    :param scheduler_options: Dictionary of additional parameters to pass to the
+        scheduler: `num_warmup_steps`, `num_training_steps`, and `scheduler_step`
+        (either `batch` or `epoch`)
```
Comment (gmastrapas) [inline finetuner/__init__.py:200]:
```suggestion
        (either `batch` or `epoch`).
```

**Review by gmastrapas — APPROVED** [review]

**Review by bwanglzu — APPROVED** [review]
LGTM!

**Review by LMMilliken — APPROVED** [review]
LGTM
